### DISCUSSION TITLE
Supplying a layout without the "layout" tag

### DISCUSSION THREAD

**Discussion by binyamin** [harttle/liquidjs#349] [Q&A] [answered]
Given a file `input.md`, how would I instruct liquid to use the layout `base.html`, without writing `{% layout 'base.html' %}` in the file `input.md`? I'd like to programmatically set the layout for a file, based on external information.

### Use Case
The most common example is with a "layout" key in the front-matter. My use case: I'm building a static-site generation workflow at [binyamin/static-site-setup](https://github.com/binyamin/static-site-setup).

### Things to Consider
- Maybe I should inject a `Token` (`TagToken`?) into the `Template` I receive from `Liquid.parse()`
- It's gotta be possible, because Eleventy + liquidjs is a thing. Although, I can't find where in the code it's setting the layout.

**Comment by harttle** [comment, marked as answer]
The layout template name can be a variable. Just set the variable in runtime.

**Reply by binyamin** [reply]
@harttle Hmm. Setting a "layout" variable in `opts.globals` has no effect for me. To clarify, I want to mimic the behavior of Jekyll and Eleventy, where setting a front-matter key (ie. global data) will set the layout. No extra tags required, even if it can be set to a variable.

**Reply by binyamin** [reply]
@harttle I hope I wasn't confusing. I'd really appreciate some advice here.

**Reply by harttle** [reply]
How about wrapping the template before passing it to liquid.parse? Like:

```js
layout = // do your front matter parse and get layout tpl
template = `{% layout "${layout}" %}\n${template}`
```

**Reply by binyamin** [reply]
That's a good suggestion. In the meantime, I've been parsing and writing the files manually. Thank you.
